gestalt-inject/src/main/java/org/terasology/context/SoftServiceLoader.java

```
@@ -0,0 +1,97 @@
+                        while (!nameIterator.hasNext()) {
+                            URL url = urls.nextElement();
+                            try (BufferedReader reader = new BufferedReader(new InputStreamReader(url.openStream()))) {
+
+                                List<String> lines = reader.lines()
+                                    .filter(line -> line.length() != 0 && line.charAt(0) != '#')
+                                    .map(line -> {
+                                        int i = line.indexOf('#');
+                                        if (i > -1) {
+                                            return line.substring(0, i);
+                                        }
+                                        return line;
+                                    }).collect(Collectors.toList());
+
+                                nameIterator = lines.listIterator();
+                            } catch (IOException e) {
+
+                            }
+                        }
+                        return true;
+                    }
+                    return false;
+                }
+
+                @Override
+                public S next() {
+                    if (!hasNext()) {
+                        throw new NoSuchElementException();
+                    }
+                    String clazz = nameIterator.next();
+                    try {
+                        return (S) classLoader.loadClass(clazz).getDeclaredConstructor().newInstance();
+                    } catch (ClassNotFoundException e) {
+                        e.printStackTrace();
+                    } catch (IllegalAccessException e) {
+                        e.printStackTrace();
+                    } catch (InstantiationException e) {
+                        e.printStackTrace();
+                    } catch (NoSuchMethodException e) {
+                        e.printStackTrace();
+                    } catch (InvocationTargetException e) {
```

my worries about swallowing all these exceptions aside, this could be more concise with `(ClassNotFoundException | IllegalAccessException` etc